agent/rpcclient/health/view.go

```diff
@@ -100,6 +111,17 @@ func (s *HealthView) Update(events []*pbsubscribe.Event) error {
 	return nil
 }
 
+func (s *HealthView) skipFilter(csn *structs.CheckServiceNode) bool {
+	// we only do this for services that need to be routed through a gateway
+	if s.kind != "" {
+		return false
+	}
+	if s.name != csn.Service.Service && csn.Service.Kind == structs.ServiceKindTerminatingGateway {
```

Comment:
I wonder if `s.name != csn.Service.Service` is guaranteed to be the case.

In theory you could name the terminating gateway the same thing as some external service (I don't think this would be good for users to do but it probably would work.

A possibly better indication would be storing the `Connect` parameter from the `ServiceSpecificRequest` in the `NewHealthView` method. Then the heuristic for skipping here would be: `s.connect && csn.Service.Kind  == structs.ServiceKindTerminatingGateway`. I think that would make more sense because you are requesting connect endpoints for a service and our indexing wouldn't return a tgw of the same name for that query. https://github.com/hashicorp/consul/blob/5deffbd95bca0bd57187aceacffbd0a7cc0f972b/agent/consul/state/catalog_schema.go#L239-L253

So the only way a connect query will return a tgw is if there are external services and in that case we cannot expect the tgw metadata to match.

Reply:
That makes sense to me -- I can update the logic, but in general it seems like we need to check:

1. We are requesting health for a service,
2. We are setting the `Connect` flag, and
3. Instead of getting a service back, we get a terminating gateway

so something like `s.kind == "" && s.connect && csn.Service.Kind == structs.ServiceKindTerminatingGateway`